### ISSUE TITLE
Instead of git tag 'weekly', use git branch 'weekly'

### ISSUE THREAD

**Issue by ottok** [rdiff-backup/rdiff-backup#1034]
In git a tag is supposed to be a permanent pointer to a specific commit. A git branch, on the other hand, is expected to be a pointer that keeps changing.

Current use of `weekly` as a git tag leads to errors like:

```
± git fetch --tags upstream 
From github.com:rdiff-backup/rdiff-backup
 ! [rejected]        weekly     -> weekly  (would clobber existing tag)
 ```

Users need to `git fetch --force` to get the new version of the tag. If it was a branch, git would not complain.

_Suggestion:_ When you develop, you don't need to use branch 'weekly'. You can simply replace `git tag weekly; git push --force --tags` with `git checkout -B weekly; git push weekly`. You won't even need a force push as the `weekly`  branch head will point to a child of the previous branch head.

**Comment by ericzolf** [comment]
I don't think it's that easy, the weekly tag is required to create automatically a release in GitHub, not for any kind of development purpose. I don't think a branch will cut the dice but I'm open to suggestions; here is the place that needs adapting: https://github.com/rdiff-backup/rdiff-backup/blob/04a332ba8865bc7613938c65eab633cc7c5b5126/.github/workflows/weekly.yml#L140

**Comment by ottok** [comment]
I don't have experience with GitHub workflows, but I am guessing that changing
> git tag -f weekly && git push -f origin weekly
to
> git checkout -B weekly && git push -f --set-upstream=origin weekly

and

> with: tag: weekly
to
> with: commit: weekly

Would keep pushing a new version of the 'weekly' branch and run the CI on every time it is pushed?

**Comment by ericzolf** [comment]
Yeah, not that simple sadly: https://github.com/ncipollo/release-action states in the notes that

> You must provide a tag either via the action input or the git ref

And I suspect that it is a GitHub limitation, where you can only create a release on a tag.

I could possibly tag on a branch, but would this solve your actual issue?

**Comment by ericzolf** [comment]
Here we are: https://github.com/orgs/community/discussions/24690 (it's not possible without tag name). So, would a tag on a branch outside of master solve your actual issue?

**Comment by TomK32** [comment]
How about the `weekly` branch and `weekly-2025-01` tags? This also allows skipping weeks if nothing at all happened.

To safe you from looking into the manpage: `date +"%Y-%W"`
